docker/index.js

```diff
@@ -2,100 +2,241 @@ const express = require('express');
-    const outputPath = formatPath([...parsedInputPath.slice(0, parsedInputPath.length - 1), "signed_data"]);
-    const result = await ain.db.ref(outputPath).setValue({
-      value: `${JSON.stringify(SDResult.data, null, 2)}`,
-      nonce: -1,
-    })
-    .catch((e) => {
-      console.error(`setValue failure:`, e);
-    });
-  
-});
+	const { transaction } = req.body;
+	const value = transaction.tx_body.operation.value;
+	const taskId = value.task_id;
+	if(cache.get(taskId)){
+		cache.ttl(taskId, 60);
+		return;
+	}
+
+	// if request is first request, set cache 
+	cache.set(taskId, true, 60);
+	
+
+	let uploadMetadataRes; // for catch error
+
+	const inputPath = transaction.tx_body.operation.ref;
+	const parsedInputPath = parsePath(inputPath);
+
+	// pre-check the output path
+	const outputPath = await formatPath([...parsedInputPath.slice(0, parsedInputPath.length - 1), "signed_data"]);
+	const errorPath = formatPath([...parsedInputPath.slice(0, parsedInputPath.length - 1), "error"]);
+
+	console.log(outputPath);
+	// init pinata sdk
+	const pinata = new pinataSDK({ pinataApiKey, pinataSecretApiKey });
+
+
+	// get generated ainft image url with with task id
+	const result = await axios.get(`${process.env.SD_INPAINTING_ENDPOINT}/tasks/${taskId}`);
+
+	const imageUrls = result.data.result;
+
+	// get image file from url
+	const imageDataResponse = await axios.get(imageUrls[value.params.index || 1], {
+		responseType: "arraybuffer",
+	});
+
+	// for test
+	const imageBuffer = fs.readFileSync(__dirname + "/test.png");
+
+	// image file to readable stream
+	const imageDataStream = new Readable();
+	imageDataStream.push(imageBuffer);
+	imageDataStream.push(null);
+
+	// upload image to pinata
+	const options = {
+		pinataMetadata: {
+			name: `${taskId}_image`,
+		},
+		pinataOptions: {
+			cidVersion: 0,
+		},
+	};
+
+	// upload image to ipfs
+	const uploadImgRes = await pinata.pinFileToIPFS(imageDataStream, options)
+		.catch(err => {
+			ain.db.ref(errorPath).setValue({
+				value: {
+					state:"Error",
+					msg:"Image upload fail. check your inforamtion of Image"
+				},
+			})
+			.then(res => console.log(res))
+			.catch((e) => {
+				console.error(`setValue failure:`, e);
+			});
+		});
+
+	// get origin metadata
+	const originMetadata = await axios.get(`https://gateway.pinata.cloud/ipfs/${value.contract_info.metadata_cid}/${value.contract_info.token_id}`, {
+		headers: {
+			'Accept': '*/*'
+		}
+	});
+
+	// metadata will be writed in ipfs
+	const metadata = {
+		attributes: originMetadata.attributes,
+		description: originMetadata.description,
+		image: originMetadata.image,
+		name: originMetadata.name,
+		namespaces: {
+			ainetwork: {
+				ain_tx: transaction.hash, // need 
+				old_metadata: value.contract_info.metadata_cid, // do not apply yet
+				updated_at: Date.now()
+			},
+		}
+	}
+	const metadataOption = {
+		pinataMetadata: {
+			name: `${taskId}_metadata`,
+		},
+		pinataOptions: {
+			cidVersion: 0,
+		},
+	}
+
+	try{
```

Comment:
Q) is there any reason to use both try-catch here and .catch() in line 98 ?

Reply:
If there are many content in catch, then i use try-catch(it is my practice).
What is your best one of the handle functions ?

Reply:
Oh i remember reason. Now trigger can not use "res" obj. (res.send is not used because AIN is not logging)
then-catch is not possible "return" for Trigger.(just return in catch) So i use try-catch, and i will fix line98 from then-catch to try-catch. 
Thank you !